src/copilot/inspect/index.ts

```diff
@@ -30,4 +44,30 @@ async function rewrite(document: TextDocument, range: Range | Selection, _contex
         }
     };
     return [action];
+}
+
+export async function waitUntilExtensionsActivated(extensionIds: string[], interval: number = 1500) {
+    const start = Date.now();
+    return new Promise<void>((resolve) => {
+        if (extensionIds.every(id => extensions.getExtension(id)?.isActive)) {
+            logger.info(`All dependent extensions [${extensionIds.join(', ')}] are activated.`);
+            return resolve();
+        }
+        const notInstalledExtensionIds = extensionIds.filter(id => !extensions.getExtension(id));
+        if (notInstalledExtensionIds.length > 0) {
+            sendInfo('java.copilot.inspection.dependentExtensions.notInstalledExtensions', { extensionIds: `[${notInstalledExtensionIds.join(',')}]` });
+            logger.info(`Dependent extensions [${notInstalledExtensionIds.join(', ')}] are not installed, setting checking interval to 10s.`);
+        } else {
+            logger.info(`All dependent extensions are installed, but some are not activated, keep checking interval ${interval}ms.`);
+        }
+        interval = notInstalledExtensionIds ? interval : 10000;
```

Comment:
According to your comment, if there are dependent extensions not installed, the checking interval is 10s.

`notInstalledExtensionIds.length ? 10000 : interval` ?

Reply:
fixed.